$NetBSD: patch-XSA-181,v 1.1 2016/08/06 10:10:10 spz Exp $

patch for XSA-181 from upstream
note this patch is only for ARM, and thus not really relevant to
this x86-only package

--- xen/arch/arm/p2m.c.orig	2016-03-23 13:57:27.000000000 +0000
+++ xen/arch/arm/p2m.c
@@ -1084,7 +1084,8 @@ void p2m_teardown(struct domain *d)
     while ( (pg = page_list_remove_head(&p2m->pages)) )
         free_domheap_page(pg);
 
-    free_domheap_pages(p2m->root, P2M_ROOT_ORDER);
+    if ( p2m->root )
+        free_domheap_pages(p2m->root, P2M_ROOT_ORDER);
 
     p2m->root = NULL;
 
